src/main/scala/io/shiftleft/fuzzyc2cpg/passes/cfgcreation/Cfg.scala

```diff
@@ -0,0 +1,186 @@
+package io.shiftleft.fuzzyc2cpg.passes.cfgcreation
+
+import io.shiftleft.codepropertygraph.generated.{EdgeTypes, nodes}
+import io.shiftleft.fuzzyc2cpg.passes.cfgcreation.Cfg.CfgEdgeType
+import io.shiftleft.passes.DiffGraph
+import org.slf4j.LoggerFactory
+
+/**
+  * A control flow graph that is under construction, consisting of:
+  *
+  * @param entryNode  the control flow graph's first node, that is,
+  *                   the node to which a CFG that appends this CFG
+  *                   should attach itself to.
+  * @param diffGraphs control flow edges between nodes of the
+  *                   code property graph.
+  * @param fringe nodes of the CFG for which an outgoing edge type
+  *               is already known but the destination node is not.
+  *               These nodes are connected when another CFG is
+  *               appended to this CFG.
+  *
+  * In addition to these three core building blocks, we store labels
+  * and jump statements that have not been resolved and may be
+  * resolvable as parent sub trees or sibblings are translated.
+  *
+  * @param labeledNodes labels contained in the abstract syntax tree
+  *                     from which this CPG was generated
+  * @param caseLabels labels beginning with "case"
+  * @param breaks unresolved breaks collected along the way
+  * @param continues unresolved continues collected along the way
+  * @param gotos unresolved gotos collected along the way
+  *
+  * */
+case class Cfg(entryNode: Option[nodes.CfgNode] = None,
+               diffGraphs: List[DiffGraph.Builder] = List(),
```

Comment:
Two things:
1. I would rename diffGraphs to cfgEdges. This would create a better readability in the rest of the code.
2. I suggest not using the DiffGraph.Builder just to store edges. As far as I see it we only concatenate the builders anyway. So why not just use `case class CfgEdge(src, dst, cfgEdgeKind)` to store the information. That way we are also sure to avoid any overhead which may be caused by the builder.